### PR TITLE
Service gracefully shutdown

### PR DESCRIPTION
**Pull request by al8n**
Add gracefully shutdown functionality, but still does not implement lifecycle for service.

### CODE REVIEW

File: Cargo.lock
Comment (danielSanchezQ) [inline Cargo.lock:1]: Was this added by mistake?

**Review by danielSanchezQ — COMMENTED** [review]
I know that implementing the full lifecycle may be complicated, but I think this could have been a good opportunity to at least try and implement the services shutting down.

File: Cargo.lock
Comment (al8n) [reply Cargo.lock:1]: Yes

**Comment by danielSanchezQ** [comment]
I think this is superseed by #27 ? Should we close it @al8n ?

**Comment by al8n** [comment]
Close to support #27

**Comment by al8n** [comment]
> I think this is superseed by #27 ? Should we close it @al8n ?

Yes

**Pull request closed without merging**